### PR TITLE
Fix infinite path region calculations

### PR DESCRIPTION
**Pull request by hughbe**
Should check for Ok, not !Ok

Fixes https://github.com/mono/libgdiplus/issues/399
Fixes https://github.com/mono/libgdiplus/issues/398
Contributes to https://github.com/mono/libgdiplus/issues/395
Contributes to #338

### CODE REVIEW

**Comment by filipnavara** [comment]
@hughbe Can you rebase please?

**Comment by hughbe** [comment]
Rebased. are the CI failures related? I really can’t seem to be able to decipher them

**Comment by filipnavara** [comment]
The test failures may or may not be related. The Windows one I don't understand really. For the macOS it seems that there's some assert hit in HarfBuzz, a subcomponent used by Pango for the text layouting.

Would you care to submit an "empty" PR to see if the same failures are hit there as well?

**Comment by filipnavara** [comment]
@monojenkins build Mono Jenkins Windows

**Comment by hughbe** [comment]
I actually don’t have a laptop on me for the next 4 weeks so doing this all purely on my phone - not sure this can be done without GitHub desktop?

**Comment by filipnavara** [comment]
Ah, no worries, I will do it :-)

**Comment by filipnavara** [comment]
HarfBuzz is broken, but the bug was fixed meanwhile (https://github.com/harfbuzz/harfbuzz/commit/883957688cc4aa4335e5131a60b3ac3b53290726). It will take some time to propagate to Homebrew I guess. Given the frequency of the updates lately not more than week though.

**Comment by filipnavara** [comment]
The other failure is expired bzip.org domain name (https://github.com/Microsoft/vcpkg/issues/4040).

**Comment by filipnavara** [comment]
@hughbe Hopefully yes. I'll try to get someone to re-run the CI and merge if it passes through.

**Comment by filipnavara** [comment]
@monojenkins build Mono Jenkins Windows

File: src/region.c [hunk 1]
```diff
@@ -191,11 +191,17 @@ gdip_is_region_empty (const GpRegion *region, BOOL allowNegative)
...
 		if (!region->tree)
 			return TRUE;
-		if (region->tree->path)
-			return (region->tree->path->count == 0);
+		if (region->tree->path) {
+			if (region->tree->path->count == 0)
+				return TRUE;
+			
+			// Open paths are empty.
+			if ((region->tree->path->types[region->tree->path->count - 1] & PathPointTypeCloseSubpath) == 0)
```
Comment (filipnavara) [inline src/region.c]: nit: Would be nice to use `if (!gdip_path_closed (region->tree->path))` here.

Comment (hughbe) [reply src/region.c]: Fixed :) /cc @marek-safar

Comment (hughbe) [reply src/region.c]: Fixed :) /cc @marek-safar

**Pull request merged**